mathesar_ui/src/3rd-party-apis/github-releases.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * GitHub's REST API gives way more info than we have here. See [docs][1]. But
+ * I've only included the bare minimum that we need.
+ *
+ * [1]: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28
+ */
+export interface GitHubRelease {
+  id: number;
+  tag_name: string;
+  published_at: string;
+  html_url: string;
+}
+
+// Toggle these lines if you want to test with a repo that has more releases.
+// const baseUrl = 'https://api.github.com/repos/microsoft/vscode';
+const baseUrl = 'https://api.github.com/repos/centerofci/mathesar';
+
+/**
+ * GitHub's documentation recommends that we set the HTTP header
+ * `X-GitHub-Api-Version` with the version of the API that we're using, but I
+ * was not able to set that header with fetch in CORS mode. I built this code
+ * using GitHub API version 2022-11-28 and I'm assuming that our needs are
+ * simplistic enough that the GitHub API won't change under us. We may need
+ * to revisit this.
+ */
+async function getOneRelease(
+  endpoint: string,
+): Promise<GitHubRelease | undefined> {
+  const response = await fetch(`${baseUrl}${endpoint}`, { mode: 'cors' });
+  if (response.status !== 200) {
+    return undefined;
+  }
+  return (await response.json()) as GitHubRelease;
+}
```

Comment:
I'd like us to use the `getAPI` method from our `requestUtils` here, and modify the method to support headers. 

Is there a reason why we use fetch here directly? If there is, I'd rather have another function within `requestUtils` that composes fetch.

Reply:
Fixed in ade9233c021bdb54b819eb76acbda052b01c14a5, using the approach we discussed on the call. I'm still calling `fetch` here, just to keep the changes as minimal as possible.